ios/inputParser/InputParser.mm

```diff
@@ -605,9 +605,11 @@ - (void)applyProcessedStyles:(NSArray *)processedStyles
                                                 params:params];
       } else if ([styleType isEqualToNumber:@([ImageStyle getStyleType])]) {
         ImageData *imgData = (ImageData *)stylePair.styleValue;
+        BOOL isLastInText =
+            styleRange.location + styleRange.length == plainTextLength;
         [((ImageStyle *)baseStyle) addImageAtRange:styleRange
                                          imageData:imgData
-                                     withSelection:NO];
+                                     withSelection:isLastInText];
```

Comment:
I'm wondering what happens if we pass the default value like:
```
<html>
<p>Some text</p>
<p><img src="test.com" width="100" height="100" /></p>
</html>
```
I think in this case it will set the selection to the end, but the selection should not be applied during the initial HTML processing or any HTML processing

Would it be better to add a check in the setValue method directly?